### PR TITLE
Adding fix for SWIM upgrade in PnP

### PR DESCRIPTION
**Pull request by abimishr**
Problem: If Image is absent in the Cisco Catalyst Center, no error is thrown.
Fix: Have added a check for the length of image_list we fetch from the API get all software details. If the length is 0, it will throw a message showing that either image is absent or is not golden tagged, making the user aware of the possible issues.
Test cases: 1. With Cisco Catalyst 9800 WLC with wrong image name (Failed)
                    2.  With Cisco Catalyst 9800 WLC with correct image name but not golden tagged (Failed)
                    3. With Cisco Catalyst 9800 WLC with correct image name and golden tagged (Passed)

### CODE REVIEW

File: plugins/modules/pnp_intent.py [hunk 1]
```diff
@@ -497,13 +506,31 @@ def get_claim_params(self):
         }
 
         if claim_params["type"] == "CatalystWLC":
+            if not (self.validated_config[0].get('static_ip')):
+                msg = "The static IP for claiming a wireless controller must be passed"
```
[the remark this answers is not in the document]
Comment (abimishr) [reply plugins/modules/pnp_intent.py]: Added

File: plugins/modules/pnp_intent.py [hunk 2]
```diff
@@ -497,13 +506,31 @@ def get_claim_params(self):
...
+                self.pnp_cred_failure(msg=msg)
+            if not (self.validated_config[0].get('subnet_mask')):
+                msg = "The subnet mask for claiming a wireless controller must be passed"
+                self.pnp_cred_failure(msg=msg)
+            if not (self.validated_config[0].get('gateway')):
+                msg = "The gateway IP for claiming a wireless controller must be of passed"
```
[the remark this answers is not in the document]
Comment (abimishr) [reply plugins/modules/pnp_intent.py]: Added

File: plugins/modules/pnp_intent.py [hunk 3]
```diff
@@ -497,13 +506,31 @@ def get_claim_params(self):
...
+                self.pnp_cred_failure(msg=msg)
+            if not (self.validated_config[0].get('ip_interface_name')):
+                msg = "The Interface Name for claiming a wireless controller must be passed"
```
[the remark this answers is not in the document]
Comment (abimishr) [reply plugins/modules/pnp_intent.py]: Added

File: plugins/modules/pnp_intent.py [hunk 4]
```diff
@@ -497,13 +506,31 @@ def get_claim_params(self):
...
+                self.pnp_cred_failure(msg=msg)
+            if not (self.validated_config[0].get('vlan_id')):
+                msg = "The Vlan Id for claiming a wireless controller must be passed"
```
[the remark this answers is not in the document]
Comment (abimishr) [reply plugins/modules/pnp_intent.py]: Added

File: plugins/modules/pnp_intent.py [hunk 5]
```diff
@@ -633,17 +660,21 @@ def get_have(self):
...
                         if self.get_site_type() != "floor":
-                            self.msg = "The site type must be specified as 'floor'\
-                                for claiming an AP"
+                            self.msg = "The site type must be specified as 'floor' for claiming an AP"
                             self.log(str(self.msg), "ERROR")
                             self.status = "failed"
                             return self
 
+                    if len(image_list) == 0:
+                        self.msg = "Either Image {0} is not present or is not tagged Golden"\
```
[the remark this answers is not in the document]
Comment (abimishr) [reply plugins/modules/pnp_intent.py]: Added

File: plugins/modules/pnp_intent.py [hunk 6]
```diff
@@ -633,17 +660,21 @@ def get_have(self):
...
+                        self.log(self.msg, "CRITICAL")
+                        self.status = "failed"
+                        return self
+
                     if len(image_list) == 1:
                         if install_mode != "INSTALL":
-                            self.msg = "Installation mode must be in \
-                            INSTALL mode to upgrade the image. Current mode is\
-                            {0}".format(install_mode)
+                            self.msg = "Installation mode must be in INSTALL mode to upgrade the image. Current mode is {0}".format(install_mode)
```
[the remark this answers is not in the document]
Comment (abimishr) [reply plugins/modules/pnp_intent.py]: Added

**Review by madhansansel — APPROVED** [review]

**Pull request merged**